src/google/protobuf/map.h

```diff
@@ -548,7 +548,7 @@ class Map {
     !defined(GOOGLE_PROTOBUF_OS_NACL) && !defined(GOOGLE_PROTOBUF_OS_ANDROID)
     template<class NodeType, class... Args>
     void construct(NodeType* p, Args&&... args) {
-      new (static_cast<void*>(p)) NodeType(std::forward<Args>(args)...);
+      new (const_cast<void*>(static_cast<const void*>(p))) NodeType(std::forward<Args>(args)...);
```

Comment:
Thanks for the explanation. Could you add it as a comment in the code (so a reader of the code can figure out where to look at to understand the context)?